scripts/py_matter_yamltests/test_yaml_parser.py

```diff
@@ -175,6 +178,81 @@
           value: TestEnum.UnknownEnumValue(0)
 '''
 
+_BASIC_ARITHMETIC_ARG_RESULTS = [6, 6, 2, 2, 8, 8, 2, 2, 0]
+basic_arithmetic_yaml = '''
+name: Test Cluster Tests
+
+config:
+    nodeId: 0x12344321
+    cluster: "Test"
+    endpoint: 1
+    myVariable: 4
+
+tests:
+    - label: "Add 2"
+      command: "IntTest"
+      arguments:
+          values:
+              - name: "arg"
+                value: myVariable + 2
+
+    - label: "Add 2 with no space"
+      command: "IntTest"
+      arguments:
+          values:
+              - name: "arg"
+                value: myVariable+2
+
+    - label: "Minus 2"
+      command: "IntTest"
+      arguments:
+          values:
+              - name: "arg"
+                value: myVariable - 2
+
+    - label: "Minus 2 with no space"
+      command: "IntTest"
+      arguments:
+          values:
+              - name: "arg"
+                value: myVariable-2
+
+    - label: "Multiply by 2"
+      command: "IntTest"
+      arguments:
+          values:
+              - name: "arg"
+                value: myVariable * 2
+
+    - label: "Multiply by 2 with no space"
+      command: "IntTest"
+      arguments:
+          values:
+              - name: "arg"
+                value: myVariable*2
+
+    - label: "Divide by 2"
+      command: "IntTest"
+      arguments:
+          values:
+              - name: "arg"
+                value: myVariable / 2
+
+    - label: "Divide by 2 with no space"
+      command: "IntTest"
+      arguments:
+          values:
+              - name: "arg"
+                value: myVariable/2
+
+    - label: "Arithmetic with parentheses"
+      command: "IntTest"
+      arguments:
+          values:
+              - name: "arg"
+                value: (myVariable -3)/ 2
```

Comment:
I would suggest `(myVariable +3)/7` evaluating to 1...  The 0 makes it look too much like something that could happen by accident as a result of error handling.